terraform_destroy_all.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# A script for destroying assessment environments created using
+# cisagov/cool-assessment-terraform.
+#
+# This is necessary because the CloudWatch alarm resources associated
+# with the EC2 instances are created using for_each expressions that
+# are dynamic-ish.
+#
+# Examples:
+# - See what would be destroyed:
+#   $ AWS_PROFILE=cool-user AWS_SHARED_CREDENTIALS_FILE=~/.aws/production_credentials AWS_DEFAULT_REGION=us-east-1 ./terraform_destroy_all.sh -var-file=envX-production.tfvars
+# - Destroy it!  (You will not be prompted.)
+#   $ AWS_PROFILE=cool-user AWS_SHARED_CREDENTIALS_FILE=~/.aws/production_credentials AWS_DEFAULT_REGION=us-east-1 ./terraform_destroy_all.sh -auto-apply -var-file=envX-production.tfvars
+
+# Export some environment variables that we want the terraform child
+# processes to inherit.
+export AWS_PROFILE
+export AWS_SHARED_CREDENTIALS_FILE
+export AWS_DEFAULT_REGION
+
+# Create a list of resources to target from the terraform state and
+# then pass them to terraform destroy.
+terraform state list | sed "s/^/-target='/;s/$/'/" | xargs terraform destroy "$@"
```

Comment:
It might be nice to expand either this comment or the one at the start of this file to add a little bit more detail about why a regular `terraform destroy` doesn't work in this case.

Reply:
I will make it so.

Reply:
Please see commit c4f7a5d.

Reply:
Looks great, thanks for expounding on that.